src/rust/src/builder_ffi.rs

```diff
@@ -100,15 +147,27 @@ pub extern "C" fn orchard_unauthorized_bundle_free(
 #[no_mangle]
 pub extern "C" fn orchard_unauthorized_bundle_prove_and_sign(
     bundle: *mut Bundle<InProgress<Unproven, Unauthorized>, Amount>,
+    keys: *const *const SpendingKey,
+    keys_len: size_t,
     sighash: *const [u8; 32],
 ) -> *mut Bundle<Authorized, Amount> {
     let bundle = unsafe { Box::from_raw(bundle) };
+    let keys = unsafe { slice::from_raw_parts(keys, keys_len) };
     let sighash = unsafe { sighash.as_ref() }.expect("sighash pointer may not be null.");
     let pk = unsafe { ORCHARD_PK.as_ref() }.unwrap();
 
+    let signing_keys = keys
+        .iter()
+        .map(|sk| {
+            unsafe { sk.as_ref() }
```

Comment:
I see; this makes sense, you're checking each pointer in the slice to ensure that they're not null individually.